src/js/components/PodView.js

```diff
@@ -0,0 +1,370 @@
+        <col style={{width: '160px'}} />
+      </colgroup>
+    );
+  }
+
+  getColumnHeading(prop, order, sortBy) {
+    let caretClassNames = classNames(
+      'caret',
+      {
+        [`caret--${order}`]: order != null,
+        'caret--visible': prop === sortBy.prop
+      }
+    );
+
+    return (
+      <span>
+        {PodTableHeaderLabels[prop]}
+        <span className={caretClassNames}></span>
+      </span>
+    );
+  }
+
+  getColumnClassName(prop, sortBy, row) {
+    return classNames({
+      'highlight': prop === sortBy.prop,
+      'clickable': row == null,
+      'table-cell-task-dot': prop === 'status'
+    });
+  }
+
+  getColumns() {
+    return [
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'name',
+        render: this.renderColumn_ID,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'address',
+        render: this.renderColumn_Address,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'status',
+        render: this.renderColumn_Status,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'cpus',
+        render: this.renderColumn_Resource,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'mem',
+        render: this.renderColumn_Resource,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'updated',
+        render: this.renderColumn_Updated,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'version',
+        render: this.renderColumn_Version,
+        sortable: true
+      }
+    ];
+  }
+
+  getTableDataFor(instances, filter) {
+    let spec = this.props.pod.getSpec();
+
+    return instances.getItems().map(function (instance) {
+      let resourcesSummary = {
+        cpus: 0, mem: 0, disk: 0, gpus: 0
+      };
+
+      let children = instance.getContainers()
+        .filter(function (container) {
+          return PodUtil.isContainerMatchingText(container, filter.text);
+        })
+        .map(function (container) {
+          let containerSpec = spec.getContainerSpec(container.name);
+          Object.keys(containerSpec.resources).forEach(function (key) {
+            resourcesSummary[key] = containerSpec.resources[key];
+          });
+
+          return {
+            id: container.getId(),
+            name: container.getName(),
+            status: container.getContainerStatus(),
+            address: container.endpoints.reduce(function (components, ep) {
```

Comment:
There are a lot of nested loops here, which makes this hard to follow.
Could we break it up a bit